### PR TITLE
[GAL-4259] include isBetaTester in events

### PR DESCRIPTION
**Pull request by kaitoo1**
### Summary of Changes
Include in events whether or not a user is a beta tester.
there are two methods: including the flag in every event, and setting the flag on the user profile.
there are pros and cons to each method, so we'll start by doing both to minimize gaps or incomplete data. we can evaluate if one turns out to be more reliable or accurate. for example, one difference is user role state - we may add or remove Beta Tester roles in the future, and including it in every event maintains historical context because you can see more directly whether a user was a beta tester at the time the event was triggered. 

### Demo or Before/After Pics

- If this is a new feature, include screenshots or recordings of the feature in action.
- If this PR makes visual changes, include before and after screenshots. You can type `/table` to quickly come up with a before/after table.

### Edge Cases

List any common edge cases that you have considered and tested.

### Testing Steps

Provide steps on how the reviewer can test the changes.

### Checklist

Please make sure to review and check all of the following:

- [ ] I've tested the changes and all tests pass.
- [ ] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [ ] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: packages/shared/src/contexts/AnalyticsContext.tsx [hunk 1]
```diff
@@ -66,22 +78,27 @@ const AnalyticsProvider = memo(({ children, identify, track }: Props) => {
...
         if (!userId) {
           return;
         }
-
         identify(userId);
+
+        if (user.roles?.includes('BETA_TESTER')) {
+          setIsBetaTester(true);
+          registerSuperProperties({ isBetaTester: true });
+        }
```
Comment (jakzaizzat) [inline packages/shared/src/contexts/AnalyticsContext.tsx:93]: beautiful 😍

File: apps/web/src/contexts/analytics/WebAnalyticsProvider.tsx [hunk 2]
```diff
@@ -41,11 +42,26 @@ export const _identify: IdentifyFunction = (userId) => {
...
 
+export const _registerSuperProperties: RegisterSuperPropertiesFunction = (eventProps) => {
+  if (!mixpanelEnabled) return;
+
+  try {
+    mixpanel.register(eventProps);
+  } catch (error: unknown) {
+    // mixpanel errors shouldn't disrupt app
+    captureException(error);
+  }
```
Comment (jakzaizzat) [inline apps/web/src/contexts/analytics/WebAnalyticsProvider.tsx:53]: Nice catch!

**Review by jakzaizzat — APPROVED** [review]
LGTM!
